klippy/webhooks.py

```diff
@@ -417,31 +407,26 @@ def _process_status_request(self, requested_objects, eventtime):
         return result
 
     def _handle_object_request(self, web_request):
-        if web_request.get_method() != 'GET':
-            raise web_request.error("Invalid Request Method")
         web_request.send(dict(self.available_objects))
 
     def _handle_status_request(self, web_request):
-        if web_request.get_method() != 'GET':
-            raise web_request.error("Invalid Request Method")
         args = web_request.get_args()
         eventtime = self.reactor.monotonic()
         result = self._process_status_request(args, eventtime)
         web_request.send(result)
 
     def _handle_subscription_request(self, web_request):
-        method = web_request.get_method()
-        if method == 'POST':
-            # add a subscription
-            args = web_request.get_args()
-            if args:
-                self.add_subscripton(args)
-            else:
-                raise web_request.error("Invalid argument")
-        else:
-            # get subscription info
-            result = dict(self.subscriptions)
-            web_request.send(result)
+        # XXX - should allow per-client subscription lists
+        args = web_request.get_args()
+        if not args:
+            raise web_request.error("Invalid argument")
+        self.add_subscripton(args)
+        cconn = web_request.get_client_connection()
+        template = web_request.get('response_template', {})
+        self.clients[cconn] = template
```

Comment:
Should we only update self.clients if the `response_template` is in the web_request?  Moonraker is always going to make an initial subscription request so it can track Klippy's state, where it can set the `response_template`.  After that, Moonraker's clients will send these requests.  

I can have Moonraker add a `response_template` to requests from its clients if necessary.